src/commands/scaffold.ts

```diff
@@ -0,0 +1,257 @@
+            default: false,
+          },
+          database: {
+            describe:
+              projectCreateRequest['project.branch.database_name'].description,
+            type: 'string',
+          },
+          role: {
+            describe:
+              projectCreateRequest['project.branch.role_name'].description,
+            type: 'string',
+          },
+          'set-context': {
+            type: 'boolean',
+            describe: 'Set the current context to the new project',
+            default: false,
+          },
+          cu: {
+            describe:
+              'The number of Compute Units. Could be a fixed size (e.g. "2") or a range delimited by a dash (e.g. "0.5-3").',
+            type: 'string',
+          },
+        }),
+      async (args) => {
+        // @ts-expect-error: TODO - Assert `args` is `CommonProps`
+        await start(args);
+      },
+    );
+};
+export const handler = (args: yargs.Argv) => {
+  return args;
+};
+
+async function getTemplateList() {
+  const content = await getContent(REPOSITORY_OWNER, REPOSITORY);
+
+  return content
+    .filter((el: any) => el.type === 'dir')
+    .map((el: any) => ({ name: el.name }));
+}
+
+const list = async (props: CommonProps) => {
+  const out = writer(props);
+
+  out.write(await getTemplateList(), {
+    fields: TEMPLATE_LIST_FIELDS,
+    title: 'Templates',
+  });
+  out.end();
+};
+
+const start = async (
+  props: CommonProps &
+    IdOrNameProps & {
+      id: string;
+      projectId?: string;
+      outputDir?: string;
+      name?: string;
+      regionId?: string;
+      cu?: string;
+      orgId?: string;
+      database?: string;
+      role?: string;
+      psql: boolean;
+      setContext: boolean;
+      '--'?: string[];
+    },
+) => {
+  const availableTemplates = (await getTemplateList()).map(
+    (el: any) => el.name,
+  );
+  if (!availableTemplates.includes(props.id)) {
+    log.error(
+      'Template not found. Please make sure the template exists and is public.',
+    );
+    return;
+  }
+  let projectData: any;
+  if (!props.projectId) {
+    projectData = await createProject(props);
+  } else {
+    projectData = (await props.apiClient.getProject(props.projectId)).data;
+    const branches: any = (
+      await props.apiClient.listProjectBranches(props.projectId)
+    ).data.branches;
+    const roles: any = (
+      await props.apiClient.listProjectBranchRoles(
+        props.projectId,
+        branches[0].id,
+      )
+    ).data.roles;
+
+    const connectionString: any = (
+      await props.apiClient.getConnectionUri({
+        projectId: props.projectId,
+        database_name: branches[0].name,
+        role_name: roles[0].name,
+      })
+    ).data.uri;
+    const connectionUrl = new URL(connectionString);
+
+    projectData.connection_uris = [
+      {
+        connection_uri: connectionString,
+        connection_parameters: {
+          database: projectData.project.name,
```

Comment:
The project name is not the database name, project names could have spaces and emoji etc. I guess you could try to parse out the path from the URL but there might be a better way.